### ISSUE TITLE
canu failed with 'failed to overlapStoreSorter

### ISSUE THREAD

**Issue by MartinPippel** [marbl/canu#214]
Hi,

we are trying to assemble a 800M genome (70x coverage, pacbio reads) on a SLURM grid engine. Unfortunately step 10 crashed with the following error message: 

`-- 3516 overlap store sorter jobs failed:`
`--   job out/correction/test.ovlStore.BUILDING/0041 FAILED.`
`--   job out/correction/test.ovlStore.BUILDING/0042 FAILED.`
`--   job out/correction/test.ovlStore.BUILDING/0043 FAILED.`
 and so forth 

`canu failed with 'failed to overlapStoreSorter.  Made 2 attempts, jobs still failed'.` 

The correction/test.ovlStore.BUILDING/log folder contains a lot files of the following error message:

`ERROR: short read on 'out/correction/test.ovlStore.BUILDING/bucket0326/sliceSizes'.`
`ERROR: read 4081 sizes insteadof 4501.`
`ovStoreSorter: stores/ovStoreSorter.C:206: int main(int, char**): Assertion nr == fileLimit + 1 failed.`

Do you have any hint what causes this crash?

**Comment by skoren** [comment]
Did all the overlap store sorting jobs fail or a subset (e.g. did you only have 3516 jobs or more)? This looks like an error from the previous step which generates the sliceSizes file at the very end of that step. Are there any errors from the logs/1-\*.log files? If there aren't, the likeliest cause is a disk issue (out of space or network issue). You can try removing the ovlStore.BUILDING folder and running Canu again and it will try re-building the store from scratch.

**Comment by MartinPippel** [comment]
Thanks for the fast reply. It took a while to rerun the jobs. You were right it was a storage/quota problem. 
Is there a possibility to guess the storage footprint for the assembly? I have a quota of 40TB, and used already 15Tb until the correction phase. I am a little concerned, that the storage is not sufficient for the full assembly. Can I clean up some intermediate files?

Another issue concerning run time limits came up. The correction step was canceled by the queuing system due to the run time limit.  

I found a way to submit the assembly to our long queue: 
`useGrid=true gridOptions="--mem 200000 -p long -t 7-12:00"`

But I do not want to submit the highly parallel jobs also to our long queue. Because of the limited number of nodes. Is there way to split the individual assembly steps to specify different queuing options?

**Comment by MartinPippel** [comment]
The default time limit is 24h. 

`ls -lrt correction`

`test.gkpStore.gkp`
`test.gkpStore.err`
`test.gkpStore`
`0-mercounts`
`test.ovlStore`
`test.ovlStore.summary.err`
`1-overlapper`
`test.ovlStore.per-read.log`
`2-correction`

last canu log:

`-- Detected Java(TM) Runtime Environment '1.8.0_71' (from '/sw/apps/jdk/current/bin/java').
-- Detected 24 CPUs and 252 gigabytes of memory.
-- Detected Slurm with 'sinfo' binary in /usr/bin/sinfo.
-- Detected Slurm with 'MaxArraySize' limited to 1000 jobs.
--
-- Found 105 hosts with  24 cores and  249 GB memory under Slurm control.
-- Found   4 hosts with  48 cores and 1005 GB memory under Slurm control.
--
-- Allowed to run under grid control, and use up to  12 compute threads and  124 GB memory for stage 'bogart (unitigger)'.
-- Allowed to run under grid control, and use up to  12 compute threads and   32 GB memory for stage 'mhap (overlapper)'.
-- Allowed to run under grid control, and use up to  12 compute threads and   32 GB memory for stage 'mhap (overlapper)'.
-- Allowed to run under grid control, and use up to  12 compute threads and   32 GB memory for stage 'mhap (overlapper)'.
-- Allowed to run under grid control, and use up to   8 compute threads and    8 GB memory for stage 'read error detection (overlap error adjustment)'.
-- Allowed to run under grid control, and use up to   1 compute thread  and    2 GB memory for stage 'overlap error adjustment'.
-- Allowed to run under grid control, and use up to   8 compute threads and   48 GB memory for stage 'utgcns (consensus)'.
-- Allowed to run under grid control, and use up to   1 compute thread  and    4 GB memory for stage 'overlap store parallel bucketizer'.
-- Allowed to run under grid control, and use up to   1 compute thread  and   16 GB memory for stage 'overlap store parallel sorting'.
-- Allowed to run under grid control, and use up to   1 compute thread  and    8 GB memory for stage 'overlapper'.
-- Allowed to run under grid control, and use up to   8 compute threads and   12 GB memory for stage 'overlapper'.
-- Allowed to run under grid control, and use up to   8 compute threads and   12 GB memory for stage 'overlapper'.
-- Allowed to run under grid control, and use up to  12 compute threads and   64 GB memory for stage 'meryl (k-mer counting)'.
-- Allowed to run under grid control, and use up to   4 compute threads and   20 GB memory for stage 'falcon_sense (read correction)'.
-- Allowed to run under grid control, and use up to  12 compute threads and   32 GB memory for stage 'minimap (overlapper)'.
-- Allowed to run under grid control, and use up to  12 compute threads and   32 GB memory for stage 'minimap (overlapper)'.
-- Allowed to run under grid control, and use up to  12 compute threads and   32 GB memory for stage 'minimap (overlapper)'.
--
-- This is canu parallel iteration #1, out of a maximum of 2 attempts.
--
-- Final error rates before starting pipeline:
--
--   genomeSize          -- 800000000
--   errorRate           -- 0.025
--
--   corOvlErrorRate     -- 0.075
--   obtOvlErrorRate     -- 0.075
--   utgOvlErrorRate     -- 0.075
--
--   obtErrorRate        -- 0.075
--
--   cnsErrorRate        -- 0.075
--
--
-- BEGIN CORRECTION
--
----------------------------------------
-- Starting command on Mon Aug  8 09:42:08 2016 with 1915255.338 GB free disk space

    /lustre/projects/dazzler/pippel/canu/Linux-amd64/bin/generateCorrectionLayouts \
      -G /lustre/projects/dazzler/pippel/canu_test/out/correction/test.gkpStore \
      -O /lustre/projects/dazzler/pippel/canu_test/out/correction/test.ovlStore \
      -S /lustre/projects/dazzler/pippel/canu_test/out/correction/2-correction/test.globalScores \
      -C 80 \
      -p /lustre/projects/dazzler/pippel/canu_test/out/correction/2-correction/test.estimate`

**Comment by brianwalenz** [comment]
Great!  Disk space peaks when building the overlap store for this stage.

It's now reading that gigantic store to configure the next (parallel) step.  This isn't fast; it's one process that needs to read the entire store.  You can make just the canu process (the "executive" process, where most of the non-parallel tasks are run) run in the long queue, and let all the parallel steps run in the normal queue:

gridOptionsExecutive="--mem 200000 -p long -t 7-12:00"

The only parallel step we've exceeded time limits on is read correction, which is the next step for you.  Increasing corPartitions might help.  The default for an 800mbp genome is 512.

**Comment by MartinPippel** [comment]
Cool. Thanks. But as its running since 12 hours I don't wanna kill it.
I can also keep it running as it is and once this step is finished successfully I will cancel it and submit the next phase in the short queue, right? 

Canu is able to detect the finished assembly phases? 
And I can just call canu like this afterwards:

`canu -d out -p test -genomeSize=800M -maxThreads=24 useGrid=true gridOptions="--mem 200000 -p short -t 0-23:54" -pacbio-raw READS.gz`

**Comment by brianwalenz** [comment]
Yes to all.

**Comment by MartinPippel** [comment]
thanks
